city/routers.py

```
@@ -0,0 +1,38 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from city import crud, schemas
+from database import AsyncSessionLocal
+
+router = APIRouter()
+
+
+async def get_db():
+    async with AsyncSessionLocal() as db:
+        yield db
+
+
+@router.get("/", response_model=list[schemas.City])
+async def read_city(db: AsyncSession = Depends(get_db)):
+    return await crud.get_all_cities(db=db)
+
+
+@router.get("/{city_id}/", response_model=schemas.City)
+async def read_single_city(city_id: int, db: AsyncSession = Depends(get_db)):
+    city = await crud.get_city(db=db, city_id=city_id)
+    if city is None:
```

The `crud.get_city` function should return a city object or raise an exception if not found. Ensure that the CRUD function is correctly implemented to handle this.